frontend/src/scenes/plugins/edit/PluginDrawer.tsx

```diff
@@ -75,14 +77,20 @@ export function PluginDrawer(): JSX.Element {
             form.setFieldsValue({
                 ...(editingPlugin.pluginConfig.config || defaultConfigForPlugin(editingPlugin)),
                 __enabled: editingPlugin.pluginConfig.enabled,
+                __matchActionId: editingPlugin.pluginConfig.match_action,
                 ...editingPluginInitialChanges,
             })
             generateApiKeysIfNeeded(form)
         } else {
             form.resetFields()
         }
         updateInvisibleAndRequiredFields()
-    }, [editingPlugin?.id, editingPlugin?.config_schema])
+    }, [editingPlugin?.id, editingPlugin?.config_schema, editingPlugin?.pluginConfig?.match_action])
+
+    const actionMatchingFlag = !!useFeatureFlag('PLUGINS_ACTION_MATCHING')
+    const actionMatchingEnabled =
+        (actionMatchingFlag || editingPlugin?.pluginConfig.match_action) &&
+        editingPlugin?.capabilities?.methods?.includes('composeWebhook')
```

Comment:
lets initially only do this for a specific plugin URL like we had in the backend